### PR TITLE
Improve readRational to support rational syntax

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

**Comment by newhoggy** [comment]
For example:

```
--price-execution-steps 22/10 --price-execution-memory 3.3
```

Will parse a rational of `11%5` and `33%10` respectively.

**Comment by newhoggy** [comment]
Fixes https://github.com/input-output-hk/cardano-node/issues/2920

**Review by erikd — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 1]
```diff
@@ -2779,8 +2780,15 @@ readRationalUnitInterval = readRational >>= checkUnitInterval
      | q >= 0 && q <= 1 = return q
      | otherwise        = fail "Please enter a value in the range [0,1]"
 
+readFractionAsRational :: Opt.ReadM Rational
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:2783]: @newhoggy can we implement a property test for this? We really want to know if this breaks.

**Review by Jimbo4350 — COMMENTED** [review]
LGTM  ~one comment~

**Review by Jimbo4350 — APPROVED** [review]

**Comment by newhoggy** [comment]
bors merge

**Comment by newhoggy** [comment]
bors merge

**Pull request merged**